app/liquid_tags/github_tag/github_code_tag.rb

```diff
@@ -0,0 +1,160 @@
+class GithubTag
+  class GithubCodeTag
+    PARTIAL = "liquids/github_code".freeze
+
+    def initialize(link)
+      @original_link = get_original_link(link)
+      @link = parse_link(link)
+      @content = get_content(@link)
+    end
+
+    def render
+      ActionController::Base.new.render_to_string(
+        partial: PARTIAL,
+        locals: {
+          file_path: @content[:file_path],
+          body: @content[:body],
+          line_info: @content[:line_info],
+          original_link: @original_link,
+          ref_name: @content[:ref_name]
+        },
+      )
+    end
+
+    def parse_link(link)
+      link = sanitize_link(link)
+      link.split(" ").first.delete(" ")
+    end
+
+    def get_repo_info(link)
+      repo_details = link.split("/")
+      raise_error if repo_details.length < 5
+
+      user_name = repo_details[0]
+      repo_name = repo_details[1]
+      ref_name = repo_details[3]
+
+      file_link = repo_details[4..-1].join("/").split("#")
+
+      {
+        user_name: user_name,
+        repo_path: "#{user_name}/#{repo_name}",
+        ref_name: ref_name,
+        file_link: file_link
+      }
+    end
+
+    def get_file_info(file_link)
+      file_path = file_link[0]
+      file_type = file_path[file_path.rindex(".") + 1..-1]
+
+      line_info = file_link[1].split("-")
+
+      raise_line_number_error unless line_info[0][0] == "L"
+      raise_line_number_error unless line_info.length == 1 || line_info[1][0] == "L"
+
+      start_line = Integer(line_info[0][1..-1])
+      end_line = if line_info.length > 1
+                   Integer(line_info[1][1..-1])
+                 end
+
+      unless end_line.nil?
+        start_line, end_line = end_line, start_line if start_line > end_line
+      end
+
+      {
+        file_path: file_path,
+        file_type: file_type,
+        start_line: start_line,
+        end_line: end_line
+      }
+    end
+
+    def get_file_contents(repo_info, file_info)
+      client = Octokit::Client.new(access_token: token)
+      file = client.contents(repo_info[:repo_path],
```

Comment:
you should use `Github::Client.contents` here. See https://github.com/thepracticaldev/dev.to/blob/3b188d349ad018d41c88ee2a9e25ca1dedac06a3/app/liquid_tags/github_tag/github_readme_tag.rb#L15